packages/fleet/lib/models/nodes.ts

```diff
@@ -0,0 +1,218 @@
+
+const NodeStateTransition = {
+    validate({ from, to }: { from: NodeState; to: Omit<NodeState, 'ERROR'> }): Result<ValidNodeStateTransitions> {
+        const transition = validNodeStateTransitions.find((t) => t.from === from && t.to === to);
+        if (transition) {
+            return Ok(transition);
+        } else {
+            return Err(new Error(`Invalid state transition from ${from} to ${to}`));
+        }
+    }
+};
+
+export interface DBNode {
+    readonly id: number;
+    readonly routing_id: RoutingId;
+    readonly deployment_id: CommitHash;
+    readonly url: string | null;
+    readonly state: NodeState;
+    readonly image: string;
+    readonly cpu_milli: number;
+    readonly memory_mb: number;
+    readonly storage_mb: number;
+    readonly error: string | null;
+    readonly created_at: Date;
+    readonly last_state_transition_at: Date;
+}
+
+export const DBNode = {
+    to: (node: Node): DBNode => {
+        return {
+            id: node.id,
+            routing_id: node.routingId,
+            deployment_id: node.deploymentId,
+            url: node.url,
+            state: node.state,
+            image: node.image,
+            cpu_milli: node.cpuMilli,
+            memory_mb: node.memoryMb,
+            storage_mb: node.storageMb,
+            error: node.error,
+            created_at: node.createdAt,
+            last_state_transition_at: node.lastStateTransitionAt
+        };
+    },
+    from: (dbNode: DBNode): Node => {
+        return {
+            id: dbNode.id,
+            routingId: dbNode.routing_id,
+            deploymentId: dbNode.deployment_id,
+            url: dbNode.url,
+            state: dbNode.state,
+            image: dbNode.image,
+            cpuMilli: dbNode.cpu_milli,
+            memoryMb: dbNode.memory_mb,
+            storageMb: dbNode.storage_mb,
+            error: dbNode.error,
+            createdAt: dbNode.created_at,
+            lastStateTransitionAt: dbNode.last_state_transition_at
+        };
+    }
+};
+
+export async function create(db: knex.Knex, nodeProps: Omit<Node, 'id' | 'createdAt' | 'state' | 'lastStateTransitionAt' | 'error'>): Promise<Result<Node>> {
+    const now = new Date();
+    const newNode: Omit<DBNode, 'id'> = {
+        routing_id: nodeProps.routingId,
+        deployment_id: nodeProps.deploymentId,
+        url: nodeProps.url,
+        state: 'PENDING',
+        image: nodeProps.image,
+        cpu_milli: nodeProps.cpuMilli,
+        memory_mb: nodeProps.memoryMb,
+        storage_mb: nodeProps.storageMb,
+        error: null,
+        created_at: now,
+        last_state_transition_at: now
+    };
+    try {
+        const inserted = await db.from<DBNode>(NODES_TABLE).insert(newNode).returning('*');
+        if (!inserted?.[0]) {
+            return Err(new Error(`Error: no node '${nodeProps.routingId}' created`));
+        }
+        return Ok(DBNode.from(inserted[0]));
+    } catch (err: unknown) {
```

Comment:
it's unknown by default
```suggestion
    } catch (err) {
```